### ISSUE TITLE
Saving a draft in ThunderBird

### ISSUE THREAD

**Issue by beroal** [roddhjav/apparmor.d#497]
ThunderBird complains that it cannot save a draft. Log:

```
apparmor="DENIED" operation="mknod" class="file" profile="thunderbird" name="/tmp/nsemail.eml"  comm="thunderbird" requested_mask="c" denied_mask="c" fsuid=1001 ouid=1001 FSUID="user" OUID="user"
```

I added to the `thunderbird` profile:

```
  @{tmp}/nsemail.eml rw,
```

Then no log messages and ThunderBird does not complain.

**Comment by roddhjav** [comment]
Thanks, should be fixed now. I recently reworked this part as it used to be `owner @{tmp}/* rw,`, we may find similar issue like this again.

**Comment by curiosityseeker** [comment]
I can confirm this issue. But there is more about /tmp : If I want to open an attachement in Thunderbird with, say, LibreOffice, Okular, whatever, Thunderbird requires write access to /tmp  - otherwise the opening of the attachment is blocked in enforce mode. I added the following lines:

```
owner /tmp/pid-@{int}*/ w,  
owner /tmp/pid-@{int}*/*.{doc,docx,pdf,PDF,png,pub,jpeg,jpg,JPG,odt,ods,rtf} w,
```

**Comment by curiosityseeker** [comment]
Adding to my above comment: This was actually my first solution. In the meanwhile I've added the following codeblock (which I'm also using in LibreOffice) below the `include <tunables/global>` line:

```
#Defines all common supported file formats
#Some obscure ones we're excluded (mostly input)

#Generic
#.txt
@{libreoffice_ext} = [tT][xX][tT]
#All the open document format
@{libreoffice_ext} += {,f,F}[oO][dDtT][tTsSpPbBgGfF]
#.xml and xsl
@{libreoffice_ext} += [xX][mMsS][lL]
#.pdf
@{libreoffice_ext} += [pP][dD][fF]
#Unified office format
@{libreoffice_ext} += [uU][oO][fFtTsSpP]
#(x)htm(l)
@{libreoffice_ext} += {,x,X}[hH][tT][mM]{,l,L}
#.epub
@{libreoffice_ext} += [eE][pP][uU][bB]
#.ps (printing to file)
@{libreoffice_ext} += [pP][sS]

#Images
@{libreoffice_ext} += [jJ][pP][gG]
@{libreoffice_ext} += [jJ][pP][eE][gG]
@{libreoffice_ext} += [pP][nN][gG]
@{libreoffice_ext} += [sS][vV][gG]
@{libreoffice_ext} += [sS][vV][gG][zZ]
@{libreoffice_ext} += [tT][iI][fF]
@{libreoffice_ext} += [tT][iI][fF][fF]

#Writer
@{libreoffice_ext} += [dD][oO][cCtT]{,x,X}
@{libreoffice_ext} += [rR][tT][fF]

#Calc
@{libreoffice_ext} += [xX][lL][sStT]{,x,X,m,M}
@{libreoffice_ext} += [xX][lL][wW]
#.dif dbf
@{libreoffice_ext} += [dD][iIbB][fF]
#.tsv .csv
@{libreoffice_ext} += [cCtT][sS][vV]
@{libreoffice_ext} += [sS][lL][kK]

#Impress/Draw
@{libreoffice_ext} += [pP][pP][tTsS]{,x,X}
@{libreoffice_ext} += [pP][oO][tT]{,m,M}
#Photoshop
@{libreoffice_ext} += [pP][sS][dD]

#Math
@{libreoffice_ext} += [mM][mM][lL]
```
and modified the second rule to:

`owner /tmp/pid-@{int}*/*.@{libreoffice_ext} w,`

I'm not sure if you're willing to add something like this.

**Comment by roddhjav** [comment]
`@{tmp}/Temp-@{uuid}/` is part of the firefox abs. To open the open the attachements, the simple solution is: `owner @{tmp}/pid-@{pid}/{,**} rw`

**Comment by curiosityseeker** [comment]
> `@{tmp}/Temp-@{uuid}/` is part of the firefox abs. 

Ah, yes, sorry.  I had obviously added that rule before the firefox abs was created in May.

> To open the open the attachements, the simple solution is: `owner @{tmp}/pid-@{pid}/{,**} w`

Sure. I wanted to explicitly restrict those attachment types. But I have to confess that I'm probably too cautious.

**Comment by roddhjav** [comment]
Should be fixed with: 89240929e9bc55a1b671766874a27af5251b369e
However, a lot of profiles may need something like `owner @{tmp}/pid-@{pid}/* r,` to open resources from tb.
